src/request/sign-transaction/SignTransaction.js

```diff
@@ -11,29 +92,63 @@ class SignTransaction { // eslint-disable-line no-unused-vars
+                fee: Nimiq.Policy.satoshisToCoins(tx.fee),
+                validityStartHeight: tx.validityStartHeight,
+                signature: signatureProof.signature.serialize(),
+                hash: tx.hash().toBase64(),
+            };
+        }
+
+        if (txRequest.type === TransactionType.EXTENDED) {
+            const key = await KeyStore.instance.get(txRequest.signer, passphraseOrPin);
+            const tx = key.createExtendedTransaction(
+                txRequest.sender,
+                txRequest.senderType,
+                txRequest.recipient,
+                txRequest.recipientType,
+                txRequest.value,
+                txRequest.fee,
+                txRequest.validityStartHeight,
+                txRequest.extraData,
+            );
+
+            const signatureProof = Nimiq.SignatureProof.unserialize(new Nimiq.SerialBuffer(tx.proof));
+
+            return {
+                type: TransactionType.EXTENDED,
+                sender: tx.sender.toUserFriendlyAddress(),
+                senderType: tx.senderType,
+                signerPubKey: signatureProof.publicKey.serialize(),
+                recipient: tx.recipient.toUserFriendlyAddress(),
+                recipientType: tx.recipientType,
+                value: Nimiq.Policy.satoshisToCoins(tx.value),
+                fee: Nimiq.Policy.satoshisToCoins(tx.fee),
+                validityStartHeight: tx.validityStartHeight,
+                signature: signatureProof.signature.serialize(),
+                hash: tx.hash().toBase64(),
+            };
         }
 
-        // eslint-disable-next-line object-curly-newline
-        const { value, fee, recipient, signer, validityStartHeight } = txRequest;
-
-        const key = await KeyStore.instance.get(signer, passphraseOrPin);
-        const tx = key.createTransaction(recipient, value, fee, validityStartHeight);
-
-        const signatureProof = Nimiq.SignatureProof.unserialize(new Nimiq.SerialBuffer(tx.proof));
-
-        return {
-            sender: tx.sender.toUserFriendlyAddress(),
-            senderPubKey: signatureProof.publicKey.serialize(),
-            recipient: tx.recipient.toUserFriendlyAddress(),
-            value: Nimiq.Policy.satoshisToCoins(tx.value),
-            fee: Nimiq.Policy.satoshisToCoins(tx.fee),
-            validityStartHeight: tx.validityStartHeight,
-            signature: signatureProof.signature.serialize(),
-            extraData: Utf8Tools.utf8ByteArrayToString(tx.data),
-            hash: tx.hash().toBase64(),
-        };
+        // The check for a valid transaction type is already done in onRequest in SignTransactionApi,
+        // but Typescript complains if this function has a path with no return.
+        // Thus logically, this should never be hit.
+        // The alternative would be to remove the check for the extended type and just run that if type is not basic.
+        throw new Error('Invalid transaction type');
```

Comment:
Probably you can avoid this if you type txRequest as union type as I did before.